src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/ErrorCodes.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime
+{
+    /// <summary>
+    ///     This class represents all of the error codes that can be
+    ///     emitted by the plugins system runtime. This class acts as a "type-safe"
+    ///     enumeration. All new explicit error conditions should be
+    ///     added to this class.
+    /// </summary>
+    public sealed class ErrorCodes
```

Comment:
Can we make the other ErrorCodes class partial, the make this another part of the class? Instead of duplicating a bunch of logic

Reply:
I.e. make both classes `public sealed partial class`

Reply:
Looks like each part of a partial class should be in the same namespace and the same assembly.